pages/docs/reference/java-to-kotlin-interop.md

```diff
@@ -397,53 +400,26 @@ public class BB8 implements Robot {
 ```
 </div>
 
-For the `@JvmDefault` annotation to take effect, the interface must be compiled with an `-Xjvm-default` argument.
-Depending on the case of adding the annotation, specify one of the argument values:
-
-* `-Xjvm-default=enabled` should be used if you add only new methods with the `@JvmDefault` annotation.
-   This includes adding the entire interface for your API.
-* `-Xjvm-default=compatibility` should be used if you are adding a `@JvmDefault` to the methods that were available in the API before.
-   This mode helps avoid compatibility breaks: all the interface implementations written for the previous versions will be fully compatible with the new version.
-   However, the compatibility mode may add some overhead to the resulting bytecode size and affect the performance.
-
-For more details about compatibility issues, see the `@JvmDefault` [reference page](/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/index.html).
-
-### Using in delegates
-
-Note that if an interface with `@JvmDefault` methods is used as a [delegate](/docs/reference/delegation.html),
-the default method implementations are called even if the actual delegate type provides its own implementations.
-
-<div class="sample" markdown="1" theme="idea" data-highlight-only>
-
-```kotlin
-interface Producer {
-    @JvmDefault fun produce() {
-        println("interface method")
-    }
-}
+### Compatibility mode for default methods
 
-class ProducerImpl: Producer {
-    override fun produce() {
-        println("class method")
-    }
-}
+If there is client Java code that uses your Kotlin interfaces compiled without `-Xjvm-default`, then recompiling it with
```

Comment:
Actually I'm not sure that we should write anything about compatibility mode in Calling Kotlin from Java' section

Reply:
Can you share your thoughts about better places for this content? 
I mean, it's about compatibility for this exact feature, hardly readers will look for this compatibility mode aside from default interfaces in general...
@udalov what do you think?

Reply:
Why not though? I think it's fine to have it here. Some users might read the section before this and decide that `-Xjvm-default=all` is a perfect choice for their project, only to find out later that they actually also needed to keep binary compatibility for some reason.

Having this section of course wouldn't completely solve this problem, but it's at least an attempt to present the full picture.

Of course, we need not delve into details about the actual problems that the all-compatibility mode has, for that we can add a link to the [blog post](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/).

Reply:
I thinks it's bad example to use `all-compatibility` mode for source compatibility cases. It's best to switch Java code to direct default methods usage - it would be more elegant and right from Java point of view. 
We also forgot that `DefaultImpls` members are deprecated in compatibility mode!
And we plan that with time `all-compatibility` mode would be subsituted with 'all' one.

So I suggest to mention `all-compatibility` mode only in case of binary compatibility issues.

Reply:
Those are good points. Maybe we should be more explicit with the fact that it only helps with binary, not source, compatibility. So instead of saying "If there is client Java code", say "If there is compiled code", and so on. And then clarify that although source compatibility is _kind of_ provided, it's recommended to rewrite Java code to use default methods, since `DefaultImpls` is going to be removed in a future version.